frontend/src/libs/components/header/header.tsx

```diff
@@ -0,0 +1,45 @@
+import profileImage from '~/assets/img/profile-image.png';
+import { Link } from '~/libs/components/components.js';
+import { useAppSelector } from '~/libs/hooks/hooks.js';
+import { type UserSignInResponseDto } from '~/packages/users/users.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  user: UserSignInResponseDto | null;
+};
+
+const Header: React.FC<Properties> = ({ user: authUserData }: Properties) => {
+  const { users } = useAppSelector(({ users }) => users);
+  const user = users.find((user) => user.id === authUserData?.id);
```

Comment:
You already have user from props
```suggestion

```

Reply:
```
type UserSignInResponseDto = {
  id: number;
  email: string;
};
```

Reply:
They may be added to response on sign in
`users` array was added for a starter as an example - it won't be used in future (especially for getting user data on a client-side))

Reply:
```
public async create(
    payload: UserSignUpRequestDto,
  ): Promise<UserSignUpResponseDto> {
    const user = await this.userRepository.create(
      UserEntity.initializeNew({
        email: payload.email,
        passwordSalt: 'SALT', // TODO
        passwordHash: 'HASH', // TODO
        firstName: payload.firstName,
        lastName: payload.lastName,
      }),
    );
```
So, if we have only sign up logic should I replace UserSignInResponseDto with UserSignUpResponseDto on the state auth user or extend UserSignInResponseDto interface?